app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -17,5 +25,302 @@ def cluster_name_href(href)
         Ovirt::Cluster.find_by_href(rhevm, href).try(:[], :name)
       end
     end
+
+    # Provisioning
+    def get_host_proxy(host, connection)
+      connection ||= ext_management_system.connect
+      host_proxy = connection.get_resource_by_ems_ref(host.ems_ref)
+      GeneralUpdateMethodNamesDecorator.new(host_proxy)
+    end
+
+    def clone_completed?(args)
+      source = args[:source]
+      phase_context = args[:phase_context]
+      logger = args[:logger]
+      # TODO: shouldn't this error out the provision???
+      return true if phase_context[:clone_task_ref].nil?
+
+      source.with_provider_connection do |rhevm|
+        status = rhevm.status(phase_context[:clone_task_ref])
+        logger.info("Clone is #{status}")
+        status == 'complete'
+      end
+    end
+
+    def destination_image_locked?(vm)
+      vm.with_provider_object do |rhevm_vm|
+        return false if rhevm_vm.nil?
+        rhevm_vm.attributes.fetch_path(:status, :state) == "image_locked"
+      end
+    end
+
+    def populate_phase_context(phase_context, vm)
+      phase_context[:new_vm_ems_ref] = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(vm[:href])
+      phase_context[:clone_task_ref] = vm.creation_status_link
+    end
+
+    def nics_for_(vm)
+      vm.with_provider_object do |rhevm_vm|
+        rhevm_vm.nics.collect { |n| NicsDecorator.new(n) }
+      end
+    end
+
+    def cluster_find_network_by_name(href, network_name)
+      ext_management_system.with_provider_connection do |rhevm|
+        Ovirt::Cluster.find_by_href(rhevm, href).try(:find_network_by_name, network_name)
+      end
+    end
+
+    def configure_vnic(args)
+      vm = args[:vm]
+      mac_addr = args[:mac_addr]
+      network = args[:network]
+      nic_name = args[:nic_name]
+      interface = args[:interface]
+      vnic = args[:vnic]
+      logger = args[:logger]
```

Comment:
I do not see any benefit to mapping most of these single-use variables.  Just reference them directory from `args` in the method.  The key names are clear.

Anything being referenced more then once it would make more sense to make a variable.